app/uk/gov/hmrc/slacknotifications/services/NotificationService.scala

```diff
@@ -106,15 +107,33 @@ class NotificationService @Inject()(
 
   private[services] def sendSlackMessage(slackMessage: SlackMessage)(
     implicit hc: HeaderCarrier): Future[NotificationResult] =
-    slackConnector.sendMessage(slackMessage).map { response =>
-      response.status match {
-        case 200 => NotificationResult().addSuccessfullySent(slackMessage.channel)
-        case _ =>
-          val slackError = SlackError(response.status, response.body)
-          Logger.warn(slackError.message)
-          NotificationResult().addError(slackError)
+    slackConnector
+      .sendMessage(slackMessage)
+      .map { response =>
+        response.status match {
+          case 200 => NotificationResult().addSuccessfullySent(slackMessage.channel)
+          case _   => logAndReturnSlackError(response.status, response.body, slackMessage.channel)
+        }
       }
-    }
+      .recover(handleSlackExceptions(slackMessage.channel))
+      .recoverWith {
+        case NonFatal(ex) =>
+          Logger.error(s"Unable to notify Slack channel ${slackMessage.channel}", ex)
+          throw ex
```

Comment:
this could be `Future.failed(ex)` to avoid rethrowing exceptions, not critical though

Reply:
Fixed in https://github.com/hmrc/slack-notifications/pull/1/commits/e20830fe9ef3ee40faec5212d2ba968a323d8935